rest-api/alembic/versions/04564bf1b5e8_deidentify_ppi_views.py

```diff
@@ -0,0 +1,279 @@
+   YEAR(qra.value_datetime) answer_datetime_year,
+   (qra.value_uri != '') answer_uri_present
+ FROM
+   participant p
+    INNER JOIN questionnaire_response qr ON p.participant_id = qr.participant_id
+    INNER JOIN questionnaire_response_answer qra
+       ON qra.questionnaire_response_id = qr.questionnaire_response_id
+    INNER JOIN questionnaire_question qq ON qra.question_id = qq.questionnaire_question_id
+    INNER JOIN questionnaire q ON qq.questionnaire_id = q.questionnaire_id
+    INNER JOIN code qc ON qq.code_id = qc.code_id
+    INNER JOIN participant_summary ps ON p.participant_id = ps.participant_id
+    LEFT OUTER JOIN hpo ON p.hpo_id = hpo.hpo_id
+    LEFT OUTER JOIN code ac ON qra.value_code_id = ac.code_id
+    WHERE (ps.email IS NULL OR ps.email NOT LIKE '%@example.com') AND
+           (hpo.name IS NULL OR hpo.name != 'TEST')
+"""
+
+_MEASUREMENTS_VIEW_SQL = """
+CREATE OR REPLACE VIEW physical_measurements_view AS
+ SELECT
+   p.participant_id participant_id,
+   YEAR(p.sign_up_time) participant_sign_up_year,
+   p.withdrawal_status participant_withdrawal_status,
+   YEAR(p.withdrawal_time) participant_withdrawal_year,
+   p.suspension_status participant_suspension_status,
+   YEAR(p.suspension_time) participant_suspension_year,
+   pm.physical_measurements_id physical_measurements_id,
+   pm.created created,
+   pm.amended_measurements_id amended_measurements_id,
+   pm.created_site_id created_site_id,
+   pm.created_username created_username,
+   pm.finalized_site_id finalized_site_id,
+   pm.finalized_username finalized_username,
+   m.measurement_id measurement_id,
+   m.code_system code_system,
+   m.code_value code_value,
+   YEAR(m.measurement_time) measurement_year,
+   m.body_site_code_system body_site_code_system,
+   m.body_site_code_value body_site_code_value,
+   (m.value_string != '') value_string_present,
+   m.value_decimal value_decimal,
+   m.value_unit value_unit,
+   m.value_code_system value_code_system,
+   m.value_code_value value_code_value,
+   YEAR(m.value_datetime) value_datetime_year,
+   m.parent_id parent_id,
+   m.qualifier_id qualifier_id
+ FROM
+   participant p
+    INNER JOIN physical_measurements pm
+       ON pm.participant_id = p.participant_id
+    INNER JOIN measurement m
+       ON pm.physical_measurements_id = m.physical_measurements_id
+    INNER JOIN participant_summary ps
+       ON p.participant_id = ps.participant_id
+    LEFT OUTER JOIN hpo
+       ON p.hpo_id = hpo.hpo_id
+  WHERE (ps.email IS NULL OR ps.email NOT LIKE '%@example.com') AND
+        (hpo.name IS NULL OR hpo.name != 'TEST') AND
+        pm.final = 1
+"""
+
+_PARTICIPANT_VIEW_SQL = """
+CREATE OR REPLACE VIEW participant_view AS
+ SELECT
+   p.participant_id,
+   YEAR(p.sign_up_time) sign_up_year,
+   p.withdrawal_status,
+   YEAR(p.withdrawal_time) withdrawal_year,
+   p.suspension_status,
+   YEAR(p.suspension_time) suspension_year,
+   hpo.name hpo,
+   IF(SUBSTR(ps.zip_code, 1, 3) IN (
```

Comment:
I'm wondering if the logic could just be if the zip < 10000?

Reply:
(For what it's worth, I thought @calbach's algorithm matches the doc)

Reply:
OK, didn't understand that. Thanks!